document/src/main/java/com/yahoo/document/json/readers/TensorRemoveUpdateReader.java

```diff
@@ -71,17 +71,25 @@ private static Tensor readRemoveUpdateTensor(TokenBuffer buffer, TensorType type
         return builder.build();
     }
 
-    private static TensorAddress readTensorAddress(TokenBuffer buffer, TensorType type) {
+    private static TensorAddress readTensorAddress(TokenBuffer buffer, TensorType type, TensorType originalType) {
         TensorAddress.Builder builder = new TensorAddress.Builder(type);
         expectObjectStart(buffer.currentToken());
         int initNesting = buffer.nesting();
         for (buffer.next(); buffer.nesting() >= initNesting; buffer.next()) {
             String dimension = buffer.currentName();
+            if ( ! type.dimension(dimension).isPresent() && originalType.dimension(dimension).isPresent()) {
+                throw new IllegalArgumentException("Indexed dimension address '" + dimension + "' should not be specified in remove update");
+            }
             String label = buffer.currentText();
             builder.add(dimension, label);
         }
         expectObjectEnd(buffer.currentToken());
         return builder.build();
     }
 
+    public static TensorType extractSparseDimensions(TensorType type) {
```

Comment:
Please move this to TensorRemoveUpdate as this detail is coupled with which tensor type (sparse) a remove update uses. Plus it is used in deserializer class which should not have a dependency to the json reader. It will also become more similar to TensorModifyUpdate.

Reply:
Done